platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -333,22 +340,33 @@ protected void releaseChunk(final Long chunk) {
         freeChunks.add(chunk);
     }
 
+    /** {@inheritDoc} */
     @Override
-    protected void partialChunkCleanup(final int chunkIndex, final long entriesToCleanUp) {
-        Long position = chunkList.get(chunkIndex);
-        if (position == null) {
-            // nothing to clean up
-            return;
-        }
+    protected void partialChunkCleanup(
+            @NonNull final Long chunkOffset, final boolean leftSide, final long entriesToCleanUp) {
         final ByteBuffer transferBuffer = initOrGetTransferBuffer();
-        transferBuffer.limit((int) (entriesToCleanUp * Long.BYTES));
-        try {
-            currentFileChannel.write(transferBuffer, position);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        fillBufferWithZeroes(transferBuffer);
+
+        if (leftSide) {
+            transferBuffer.limit((int) (entriesToCleanUp * Long.BYTES));
```

Comment:
Code would be slightly more readable, if transferBuffer position is set to 0 in this branch.

Reply:
Agreed.